src/app/searchResults/searchResults.component.js

```diff
@@ -12,29 +12,52 @@ let componentName = 'searchResults';
 class SearchResultsController {
 
   /* @ngInject */
-  constructor(designationsService) {
+  constructor($window, $location, designationsService) {
+    this.$window = $window;
+    this.$location = $location;
     this.designationsService = designationsService;
+    this.searchParams = {};
   }
 
   $onInit(){
-    let searchTerm = urlQueryTerm.get('q');
-    if(searchTerm){
-      this.designationsService.createSearch(searchTerm)
-        .subscribe((id) => {
-          this.searchId = id;
-          this.getPageResults(1);
+    var params = this.$location.search();
+
+    this.searchParams.keyword = params.q;
+    this.searchParams.first_name = params.fName;
+    this.searchParams.last_name = params.lName;
+    this.searchParams.type = params.type;
+    this.searchParamsInitial = angular.copy(this.searchParams);
+
+    this.requestSearch();
+  }
+
+  requestSearch(type){
+    if(angular.isDefined(type)){
```

Comment:
Is `requestSearch()` called by anything other than `$onInit` when `this.searchParams` is set? Do you think this first block of logic would make more sense with what you currently have in `$onInit`? I'm a little confused why it seems like you are copying the first and last name into initial and then back.